src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php

```diff
@@ -148,29 +149,48 @@ protected function getEffectivePlugins(array $resources): array
      */
     protected function extractEffectivePlugins($effectivePlugins, $pluginsPerExporter): array
     {
-        foreach ($effectivePlugins as $effectivePlugin) {
-            if ($effectivePlugin instanceof EventResourceRepositoryPluginInterface || $effectivePlugin instanceof EventResourceBulkRepositoryPluginInterface) {
-                $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS][$effectivePlugin->getResourceName()][$effectivePlugin->getEventName()] = $effectivePlugin;
+        foreach ($effectivePlugins as $effectiveEventPlugins) {
+            $effectivePlugin = $this->resolveDuplicatePlugins($effectiveEventPlugins);
+
+            if ($this->isEventResourceRepositoryPlugin($eventResourcePlugin)) {
+                $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS][] = $effectivePlugin;
             }
 
             if ($effectivePlugin instanceof EventResourceQueryContainerPluginInterface) {
-                $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS][$effectivePlugin->getResourceName()][$effectivePlugin->getEventName()] = $effectivePlugin;
+                $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS][] = $effectivePlugin;
             }
         }
 
-        $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS] = $this->resolveDuplicatePlugins($pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS]);
-        $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS] = $this->resolveDuplicatePlugins($pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS]);
-
         return $pluginsPerExporter;
     }
 
     /**
      * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[] $eventResourcePlugins
      *
-     * @return \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[]
+     * @return \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface|null
+     */
+    protected function resolveDuplicatePlugins(array $eventResourcePlugins): ?EventResourcePluginInterface
+    {
+        if ($eventResourcePlugins === []) {
+            return null;
+        }
+
+        foreach ($eventResourcePlugins as $eventResourcePlugin) {
+            if ($this->isEventResourceRepositoryPlugin($eventResourcePlugin)) {
+                return $eventResourcePlugin;
+            }
+        }
+
+        return current($eventResourcePlugins);
+    }
+
+    /**
+     * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface $plugin
+     *
+     * @return bool
      */
-    protected function resolveDuplicatePlugins(array $eventResourcePlugins): array
+    protected function isEventResourceRepositoryPlugin(EventResourcePluginInterface $plugin): bool
```

Comment:
Please add nullable support , on line 155 in `$eventResourcePlugin` variable could be null based on code.

Reply:
Fixed.